src/rgw/driver/rados/rgw_d3n_datacache.cc

```diff
@@ -341,7 +354,7 @@ size_t D3nDataCache::random_eviction()
     d3n_cache_map.erase(del_oid); // oid
   }
 
-  location = cache_location + url_encode(del_oid, true);
+  location = cache_location + del_oid;
   ::remove(location.c_str());
```

Comment:
Hi @mkogan1  Do we need to check if the parent directory and its ancestors are empty and remove them if they are? lru_eviction as same.